bulldozer/config_v1.go

```diff
@@ -47,6 +47,8 @@ type MergeConfig struct {
 	Method  MergeMethod                 `yaml:"method"`
 	Options map[MergeMethod]MergeOption `yaml:"options"`
 
+	BranchMergeMethod map[string]MergeMethod `yaml:"branch_merge_method"`
```

Comment:
Since this appears in the `merge` block, I'd like to call it `BranchMethod` / `branch_method` to better align with the default `method` property.